lms/djangoapps/mobile_api/middleware.py

```diff
@@ -0,0 +1,78 @@
+"""
+Middleware for Mobile APIs
+"""
+from django.conf import settings
+from django.http import HttpResponse
+import re
+from mobile_api.models import MobileAppVersionConfig
+from openedx.core.lib.mobile_utils import is_request_from_mobile_app
+
+
+class AppVersionUpgrade(object):
+    """
+    Middleware class to keep track of mobile application version being used
+    """
+    LATEST_VERSION = "EDX-APP-LATEST-VERSION"
+    UPGRADE_DEADLINE = "EDX-APP-UPGRADE-DATE"
+    MOBILE_PLATFORM_USER_AGENT_REGEX = {
+        MobileAppVersionConfig.ANDROID: (r'Dalvik/[.0-9]+ \(Linux; U; Android [.0-9]+; '
+                                         r'(.*) Build/[0-9a-zA-Z]*\) (.*)/[0-9]+.[0-9]+.[0-9]+(.[0-9a-zA-Z]*)?'),
+        MobileAppVersionConfig.IOS: (r'\([0-9]+.[0-9]+.[0-9]+(.[0-9a-zA-Z]*)?; '
+                                     r'OS Version [0-9.]+ \(Build [0-9a-zA-Z]*\)\)'),
+    }
+    MOBILE_PLATFORM_VERSION = {
```

Comment:
Regardless, creating a class abstraction for this should simplify the code and make it more readable.